node/forest_libp2p/bitswap/tests/request_manager.rs

```diff
@@ -0,0 +1,204 @@
+                *block_exist.cid(),
+                TIMEOUT,
+                Some(request_tx),
+            );
+            let success =
+                tokio::task::spawn_blocking(move || request_rx.recv_timeout(TIMEOUT)).await??;
+            assert!(success);
+            assert!(client_store.contains(block_exist.cid())?);
+        }
+
+        Ok(())
+    }
+
+    async fn create_swarm() -> Result<(Swarm<BitswapBehaviour>, PeerId, Multiaddr)> {
+        let id_keys = Keypair::generate_ed25519();
+        let peer_id = PeerId::from(id_keys.public());
+        let transport = tcp::tokio::Transport::default()
+            .upgrade(core::upgrade::Version::V1)
+            .authenticate(noise::NoiseAuthenticated::xx(&id_keys)?)
+            .multiplex(yamux::YamuxConfig::default())
+            .timeout(TIMEOUT)
+            .boxed();
+        let behaviour = BitswapBehaviour::new(&[b"/test/ipfs/bitswap/1.0.0"], Default::default());
+        let mut swarm = Swarm::with_tokio_executor(transport, behaviour, peer_id);
+        swarm.listen_on(LISTEN_ADDR.parse()?)?;
+        let peer_addr = loop {
+            let event = swarm.select_next_some().await;
+            if let SwarmEvent::NewListenAddr { address, .. } = event {
+                break address;
+            }
+        };
+
+        Ok((swarm, peer_id, peer_addr))
+    }
+
+    async fn run_swarm_loop(swarm: Swarm<BitswapBehaviour>, store: TestStore) -> Result<()> {
+        let request_manager = swarm.behaviour().request_manager();
+        let mut outbound_request_rx_stream = request_manager.outbound_request_rx().stream().fuse();
+        let mut swarm_stream = swarm.fuse();
+
+        loop {
+            select! {
+                // Hook libp2p swarm events
+                swarm_event_opt = swarm_stream.next() => {
+                    // `store` implements `BitswapStoreRead`
+                    _ = handle_swarm_event(
+                        swarm_stream.get_mut(),
+                        swarm_event_opt,
+                        store.as_ref(),
+                    );
+                },
+                request_opt = outbound_request_rx_stream.next() => if let Some((peer, request)) = request_opt {
+                    swarm_stream.get_mut().behaviour_mut().send_request(&peer, request);
+                },
+            }
+        }
+    }
+
+    fn handle_swarm_event(
+        swarm: &mut Swarm<BitswapBehaviour>,
+        swarm_event_opt: Option<
+            SwarmEvent<BitswapBehaviourEvent, <<<BitswapBehaviour as NetworkBehaviour>::ConnectionHandler as IntoConnectionHandler>::Handler as ConnectionHandler>::Error>,
+        >,
+        store: &impl BitswapStoreRead,
+    ) -> Result<()> {
+        match swarm_event_opt {
+            Some(SwarmEvent::Behaviour(event)) => {
+                let bitswap = &mut swarm.behaviour_mut();
+                bitswap.handle_event(store, event)?;
+            }
+            Some(_e) => {
+                // eprintln!("{_e:?}");
```

Comment:
Let's not have commented-out code if possible
```suggestion
```

Reply:
Fixed